modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -870,7 +870,12 @@ bool CvCapture_FFMPEG::open( const char* _filename )
             int enc_width = enc->width;
             int enc_height = enc->height;
 
-            AVCodec *codec = avcodec_find_decoder(enc->codec_id);
+            AVCodec *codec;
+            if(av_dict_get(dict, "video_codec", NULL, 0) == NULL) {
+                codec = avcodec_find_decoder(enc->codec_id);
+            } else {
+                codec = avcodec_find_decoder_by_name(av_dict_get(dict, "video_codec", NULL, 0)->value);
```

Comment:
What ffmpeg/avcodec versions are allowed to do this?

Reply:
To do what ? Find a decoder with its name ?
I will try to find this information.

Reply:
We need to support multiple version of ffmpeg (at least actual versions).

> since the v0.5 (2009) 

It is fine. Thank you!

Reply:
My pleasure !